packages/flutter/test/widgets/overscroll_stretch_indicator_test.dart

```diff
@@ -770,4 +774,238 @@ void main() {
     await pointer4.up();
     await tester.pumpAndSettle();
   });
+
+  testWidgets('Stretch overscroll vertically, change direction mid scroll', (WidgetTester tester) async {
+    final GlobalKey box1Key = GlobalKey();
+    final GlobalKey box2Key = GlobalKey();
+    final GlobalKey box3Key = GlobalKey();
+    final ScrollController controller = ScrollController();
+    await tester.pumpWidget(
+      buildTest(
+        box1Key,
+        box2Key,
+        box3Key,
+        controller,
+        // Setting the `boxHeight` to 100.0 will make the boxes fit in the
+        // scrollable viewport.
+        boxHeight: 100,
+        // To make the scroll view in the test still scrollable, we need to add
+        // the `AlwaysScrollableScrollPhysics`.
+        physics: const AlwaysScrollableScrollPhysics(),
+      ),
+    );
+
+    expect(find.byType(StretchingOverscrollIndicator), findsOneWidget);
+    expect(find.byType(GlowingOverscrollIndicator), findsNothing);
+    final RenderBox box1 = tester.renderObject(find.byKey(box1Key));
+    final RenderBox box2 = tester.renderObject(find.byKey(box2Key));
+    final RenderBox box3 = tester.renderObject(find.byKey(box3Key));
+
+    expect(controller.offset, 0.0);
+    expect(box1.localToGlobal(Offset.zero), Offset.zero);
+    expect(box2.localToGlobal(Offset.zero), const Offset(0.0, 100.0));
+    expect(box3.localToGlobal(Offset.zero), const Offset(0.0, 200.0));
+
+    final TestGesture gesture = await tester.startGesture(tester.getCenter(find.byType(CustomScrollView)));
+    // Overscroll the start
```

Comment:
Nit: missing period at the end of the sentence here and in a few other comments below.